source/docs/appendices/troubleshooting.md

```diff
@@ -0,0 +1,84 @@
+---
+id: troubleshooting
+title: Troubleshooting
+---
+
+Over the years, many developers have used Front-Commerce. A learning curve for such a solution exist, and we have detected recurring errors that may be difficult to troubleshoot for developers new to Front-Commerce or starting a new project.
+
+This page contains the most common errors you may encounter, along with information that will help to solve them quickly.
+
+## My server is not starting
+
+> I cannot prepare the application or make it start
+
+1. check the stdout/stderr of your server and the `/logs` folder, there may be useful information
+2. try to prefix your command with `DEBUG="front-commerce:scripts"` to enter verbose mode (e.g: `DEBUG="front-commerce:scripts" npm run start`)
+3. try to remove your `.front-commerce` and `node_modules` directories, run `npm install` and try again… `¯\_(ツ)_/¯`
+
+## Authentication issue
+
+> I am not logged in after entering valid credential information in the login form
+
+1. check the stdout/stderr of your server and the `/logs` folder, there may be useful information
+2. have you cleared your cookies?
+3. does the response send you a cookie with a valid domain?
+4. are secrets for your backend correctly set?
+5. is the session store correctly set? (see `config/sessions.js`)
+    * for Filesystem storage: ensure you have the permissions to write session files (the default location is `.front-commerce/sessions`)
+    * using another storage: is its configuration valid?
+    * having multiple Front-Commerce processes: ensure that you are using a session store that's compatible with a distributed architecture (ex: redis, memcached, etc.).
+
+## Redirection loop
+
+> When visiting any page, I get redirected infinitely
+
+1. check the stdout/stderr of your server and the `/logs` folder, there may be useful information
+2. what is the output of a `curl -I http://example.com` (replace with the url causing issues)
+3. it could be a redirection from the shop detection mechanism that redirects to the default one because it cannot find one matching the current url
+    1. turn on the configuration debug mode using `DEBUG="experimental:front-commerce:config"`
+    2. is the list of `validUrls` correct? Is `FRONT_COMMERCE_URL` in the `validUrls` logged?
```

Comment:
Rather than FRONT_COMMERCE_URL, it is more important to check the urls in `config/stores.js`.